File: Programming_Fundamentals_May_2016/Training/01. Programming Fundamentals Sample Exam - April 2016/VehiclePark.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _04.Vehicle_Park
{
    class VehiclePark
    {
        static void Main(string[] args)
        {
            List<string> park = Console.ReadLine().Split().ToList();

            uint sold = 0;
            string input = Console.ReadLine();

            while (input != "End of customers!")
            {
                string[] request = input.ToLower().Split();
                char requestType = request[0][0];
                uint requestSeats = uint.Parse(request[2]);

                bool foundVehicle = false;
                for (int i = 0; i < park.Count; i++)
                {
                    uint availableSeats = uint.Parse(park[i].Substring(1));
                    if (requestType == park[i][0] && requestSeats == availableSeats)
                    {
                        sold++;
                        decimal price = requestType * (decimal)requestSeats;
                        Console.WriteLine($"Yes, sold for {price}$");
                        park.Remove(park[i]);
                        foundVehicle = true;
                        break;
                    }
                }
                if (!foundVehicle)
                {
                    Console.WriteLine("No");
                }
                input = Console.ReadLine();
            }
            
            Console.WriteLine("Vehicles left: " + string.Join(", ", park));

            Console.WriteLine($"Vehicles sold: {sold}");

        }
    }
}

File: Java_Fundamentals_January_2017/Java_Advanced/Exersices/01. JavaSyntax/src/VehiclePark.java
import java.util.ArrayList;
import java.util.Scanner;

public class VehiclePark {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String[] input = scan.nextLine().split(" ");
        ArrayList<String> park = new ArrayList<>();

        if (!input[0].equals("")){
            for (String in :
                    input) {
                park.add(in);
            }
        }

        int soldVehicles = 0;
        long price;

        String line = scan.nextLine();
        while (!line.equals("End of customers!")){
            String[] order = line.toLowerCase().split(" ");

            char vehicle = order[0].charAt(0);
            int seatsCount = Integer.parseInt(order[2]);
            boolean sold = false;

            for (int i = 0; i < park.size(); i++) {
                char vehicleAvailable = park.get(i).charAt(0);
                int seatsAvailable = Integer.parseInt(park.get(i).substring(1));

                if (vehicle == vehicleAvailable && seatsCount == seatsAvailable){
                    price = seatsCount * vehicle;
                    System.out.printf("Yes, sold for %d$\n", price);
                    park.remove(i);
                    soldVehicles++;
                    sold = true;
                    break;
                }

            }

            if (!sold){
                System.out.println("No");
            }
            line = scan.nextLine();
        }

        System.out.print("Vehicles left: ");

        if (park.size() > 0){
            for (int i = 0; i < park.size() - 1; i++) {
                System.out.printf("%s, ", park.get(i));
            }
            System.out.println(park.get(park.size() - 1));
        }
        else{
            System.out.println();
        }

        System.out.printf("Vehicles sold: %d", soldVehicles);
    }
}
